tests/api/test_variant_availability.py

```diff
@@ -0,0 +1,122 @@
+from unittest.mock import Mock
+
+from saleor.product.models import ProductVariant
+from saleor.product.utils.availability import get_variant_availability
+from tests.api.utils import get_graphql_content
+
+QUERY_GET_VARIANT_AVAILABILITY = """
+query {
+  products(first: 1) {
+    edges {
+      node {
+        variants {
+          availability {
+            available
+            onSale
+            
+            discount {
+              currency
+              net {
+                amount
+              }
+            }
+            
+            priceUndiscounted {
+              currency
+              net {
+                amount
+              }
+            }
+            
+            price {
+              currency
+              net {
+                amount
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+}
+"""
+
+
+def test_get_variant_availability_on_sale(api_client, sale, product):
+    response = api_client.post_graphql(QUERY_GET_VARIANT_AVAILABILITY, {})
+    content = get_graphql_content(response)
+
+    availability = (
+        content['data']['products']['edges'][0]['node']
+        ['variants'][0]['availability'])
+
+    # ensure the availability was correctly retrieved and sent
+    assert availability
+
+    # check availability
+    assert availability['available'] is True
+    assert availability['onSale'] is True
+
+    # check the discount
+    assert availability['discount']['currency'] == 'USD'
```

Comment:
I suggest to use proper o object properties (`product.price.currency` or whatever it is) rather than hardcoded values - price in the fixture can change, same with currency, etc., it should not break tests in such case.